src/Paramore.Brighter/CommandProcessor.cs

```diff
@@ -461,6 +472,166 @@ public async Task PublishAsync<T>(
+            
+            var span = _tracer?.CreateSpan(CommandProcessorSpanOperation.Scheduler, request, requestContext?.Span, options: _instrumentationOptions);
+            var context = InitRequestContext(span, requestContext);
+
+            try
+            {
+                Message message = s_mediator!.CreateMessageFromRequest(request, context);
+
+                var scheduler = _messageSchedulerFactory.Create(this);
+                return scheduler switch
+                {
+                    IAmAMessageSchedulerSync sync => sync.Schedule(message, delay),
+                    IAmAMessageSchedulerAsync async => BrighterAsyncContext.Run(async () => await async.ScheduleAsync(message, delay)),
+                    _ => throw new InvalidOperationException("Message scheduler must be sync or async")
+                };
+            }
+            catch (Exception e)
+            {
+                _tracer?.AddExceptionToSpan(span, [e]);
+                throw;
+            }
+            finally
+            {
+                _tracer?.EndSpan(span);
+            }
+        }
+
+        /// <inheritdoc />
+        public string SchedulerPost<TRequest>(TRequest request, 
+            DateTimeOffset at, 
+            RequestContext? requestContext = null,
+            Dictionary<string, object>? args = null) where TRequest : class, IRequest
+        { 
+            if (_messageSchedulerFactory == null)
+            {
+                throw new InvalidOperationException("No message scheduler factory defined.");
+            }
+            
+            s_logger.LogInformation("Scheduling a request: {RequestType} {Id}", request.GetType(), request.Id);
+            
+            var span = _tracer?.CreateSpan(CommandProcessorSpanOperation.Scheduler, request, requestContext?.Span, options: _instrumentationOptions);
+            var context = InitRequestContext(span, requestContext);
+
+            try
+            {
+                Message message = s_mediator!.CreateMessageFromRequest(request, context);
+
+                var scheduler = _messageSchedulerFactory.Create(this);
+                return scheduler switch
+                {
+                    IAmAMessageSchedulerSync sync => sync.Schedule(message, at),
+                    IAmAMessageSchedulerAsync async => BrighterAsyncContext.Run(async () => await async.ScheduleAsync(message, at)),
+                    _ => throw new InvalidOperationException("Message scheduler must be sync or async")
+                };
+            }
+            catch (Exception e)
+            {
+                _tracer?.AddExceptionToSpan(span, [e]);
+                throw;
+            }
+            finally
+            {
+                _tracer?.EndSpan(span);
+            }
+        }
+        
+        /// <inheritdoc />
+        public async Task<string> SchedulerPostAsync<TRequest>(TRequest request, 
```

Comment:
The same comments apply. Overload with an At parameter over a new method convention; only one scheduler type

Reply:
We cover Post here, but we would also want to cover Send and Publish. Those take a Request, not a message. You might find it easier to use a Request to be pushed to the Job. That means that when your QuartzJob or the like calls back, it can just replay the request. Of course, this does not help with messaging providers, who need to support a message, as they have a message by that point. But their callback is just to reproduce their message